pkg/cmd/server/origin/master_config.go

```diff
@@ -415,6 +389,182 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
+			return nil, fmt.Errorf("error parsing master ca file for Service Account Token Manager: %s: %v", options.ServiceAccountConfig.MasterCA, err)
+		}
+	}
+	if options.ControllerConfig.ServiceServingCert.Signer != nil && len(options.ControllerConfig.ServiceServingCert.Signer.CertFile) > 0 {
+		certFile := options.ControllerConfig.ServiceServingCert.Signer.CertFile
+		serviceServingCA, err := ioutil.ReadFile(certFile)
+		if err != nil {
+			return nil, fmt.Errorf("error reading ca file for Service Serving Certificate Signer: %s: %v", certFile, err)
+		}
+		if _, err := crypto.CertsFromPEM(serviceServingCA); err != nil {
+			return nil, fmt.Errorf("error parsing ca file for Service Serving Certificate Signer: %s: %v", certFile, err)
+		}
+
+		// if we have a rootCA bundle add that too.  The rootCA will be used when hitting the default master service, since those are signed
+		// using a different CA by default.  The rootCA's key is more closely guarded than ours and if it is compromised, that power could
+		// be used to change the trusted signers for every pod anyway, so we're already effectively trusting it.
+		if len(ret.ServiceAccountTokenControllerOptions.RootCA) > 0 {
+			ret.ServiceAccountTokenControllerOptions.ServiceServingCA = append(ret.ServiceAccountTokenControllerOptions.ServiceServingCA, ret.ServiceAccountTokenControllerOptions.RootCA...)
+			ret.ServiceAccountTokenControllerOptions.ServiceServingCA = append(ret.ServiceAccountTokenControllerOptions.ServiceServingCA, []byte("\n")...)
+		}
+		ret.ServiceAccountTokenControllerOptions.ServiceServingCA = append(ret.ServiceAccountTokenControllerOptions.ServiceServingCA, serviceServingCA...)
+	}
+
+	ret.ServiceAccountControllerOptions = origincontrollers.ServiceAccountControllerOptions{
+		ManagedNames: options.ServiceAccountConfig.ManagedNames,
+	}
+
+	storageVersion := options.EtcdStorageConfig.OpenShiftStorageVersion
+	groupVersion := schema.GroupVersion{Group: "", Version: storageVersion}
+	annotationCodec := kapi.Codecs.LegacyCodec(groupVersion)
+
+	imageTemplate := variable.NewDefaultImageTemplate()
+	imageTemplate.Format = options.ImageConfig.Format
+	imageTemplate.Latest = options.ImageConfig.Latest
+
+	ret.BuildControllerConfig = origincontrollers.BuildControllerConfig{
+		DockerImage:           imageTemplate.ExpandOrDie("docker-builder"),
+		STIImage:              imageTemplate.ExpandOrDie("sti-builder"),
+		AdmissionPluginConfig: options.AdmissionConfig.PluginConfig,
+		Codec: annotationCodec,
+	}
+
+	vars, err := getOpenShiftClientEnvVars(options)
+	if err != nil {
+		return nil, err
+	}
+	ret.DeployerControllerConfig = origincontrollers.DeployerControllerConfig{
+		ImageName:     imageTemplate.ExpandOrDie("deployer"),
+		Codec:         annotationCodec,
+		ClientEnvVars: vars,
+	}
+	ret.DeploymentConfigControllerConfig = origincontrollers.DeploymentConfigControllerConfig{
+		Codec: annotationCodec,
```

Comment:
did we lost todo about moving codec to controller context? (maybe i never added that todo ;-)